test/configurationTest.ts

```diff
@@ -58,34 +126,560 @@ describe("configuration", () => {
+
+            // ATOMIST_CONFIG_PATH
+            save.ATOMIST_CONFIG_PATH = process.env.ATOMIST_CONFIG_PATH;
+            const atomistConfigPath: Configuration = {
+                cluster: {
+                    enabled: true,
+                    workers: 2,
+                },
+                ws: {
+                    enabled: true,
+                    termination: {
+                        graceful: true,
+                        gracePeriod: 3000,
+                    },
+                },
+            };
+            const atomistConfigPathFile = tmp.fileSync();
+            fs.writeJSONSync(atomistConfigPathFile.name, atomistConfigPath);
+            process.env.ATOMIST_CONFIG_PATH = atomistConfigPathFile.name;
+
+            // ATOMIST_CONFIG
+            save.ATOMIST_CONFIG = process.env.ATOMIST_CONFIG;
+            const atomistConfig: Configuration = {
+                ws: {
+                    enabled: true,
+                    compress: true,
+                    termination: {
+                        graceful: true,
+                        gracePeriod: 30,
+                    },
+                },
+            };
+            process.env.ATOMIST_CONFIG = JSON.stringify(atomistConfig);
+
+            // ATOMIST_TEAMS
+            save.ATOMIST_TEAMS = process.env.ATOMIST_TEAMS;
+            process.env.ATOMIST_TEAMS = "T61,HELPMEMARY,GLORY";
+
+            // ATOMIST_TOKENS
+            save.ATOMIST_TOKEN = process.env.ATOMIST_TOKEN;
+            process.env.ATOMIST_TOKEN = "lizphairexileinguyville";
+
+            loadConfiguration(atomistConfigJsFile.name)
+                .then(c => {
+                    const e = _.cloneDeep(defCfg);
+                    e.endpoints.graphql = "https://user.graphql.ep:1313/gql/team";
+                    e.endpoints.api = "https://user.api.ep:4141/reg";
+                    e.environment = "env-module-load";
+                    e.application = "app-module-load";
+                    e.policy = "durable";
+                    e.http.enabled = false;
+                    e.http.port = 1818;
+                    e.http.host = "atm-cfg-js";
+                    e.ws.compress = true;
+                    e.ws.termination.graceful = true;
+                    e.ws.termination.gracePeriod = 30;
+                    e.cluster.enabled = true;
+                    e.cluster.workers = 2;
+                    e.teamIds = ["T61", "HELPMEMARY", "GLORY"];
+                    e.token = "lizphairexileinguyville";
+                    e.custom = { test: "123456" };
+
+                    assert.deepStrictEqual(c, e);
+
+                    _.forEach(save, (v, k) => {
+                        if (v) {
+                            process.env[k] = v;
+                        } else {
+                            delete process.env[k];
+                        }
+                    });
+                    done();
+                });
```

Comment:
Instead of the `done()` in the line above, there should be a `.then(() => done(), done)` at the end. Or, since we can use async/await now, we could do that.

Reply:
Yeah, that’s right. But wouldn’t `.then(done, done)` also work?